app/session.go

```diff
@@ -34,10 +35,30 @@ func (a *App) CreateSession(session *model.Session) (*model.Session, *model.AppE
 	return session, nil
 }
 
+var UserSessionPool = sync.Pool{
+	New: func() interface{} {
+		return &model.Session{}
+	},
+}
+
 func (a *App) GetSession(token string) (*model.Session, *model.AppError) {
 	metrics := a.Metrics()
 
-	var session *model.Session
+	var session = UserSessionPool.Get().(*model.Session)
+	session.Id = ""
+	session.Token = ""
+	session.CreateAt = 0
+	session.ExpiresAt = 0
+	session.LastActivityAt = 0
+	session.UserId = ""
+	session.DeviceId = ""
+	session.Roles = ""
+	session.IsOAuth = false
+	session.ExpiredNotify = false
+	session.Props = nil
+	session.TeamMembers = nil
+	session.Local = false
+
 	var err *model.AppError
 	if err := a.Srv().sessionCache.Get(token, &session); err == nil {
```

Comment:
We are still passing pointer to pointer here. From our last discussion, I think we decided to pass just pointer and remove the `**` case in lru.go.

Reply:
sorry, I'm doing the changes still, have run into a small tumbling block it errors when trying to create a new account with these changes, I'm troubleshooting it at the moment.  Will close this PR and reopen when ready to review.

Reply:
Sure, no worries. Feel free to ask for my review whenever you are ready. I wasn't sure if I should have reviewed or not since you put me as a reviewer.

And there is no need to close the PR. You can just continue working on it.

Reply:
@agnivade I have re-opened the PR and is ready for your review.  I have kept it in draft mode for the moment.  Let me know if its missing anything.